python/tests/unit/processes/local_runtime/test_local_step.py

```diff
@@ -0,0 +1,558 @@
+        await step.handle_message(msg)
+
+    assert "Function mismatched_func not found in plugin my_step" in str(exc.value)
+
+
+async def test_handle_message_raises_function_not_found_if_no_function():
+    """Test handle_message raises ProcessFunctionNotFoundException if the function is not found in the step."""
+    mock_kernel = AsyncMock(spec=Kernel)
+    mock_kernel.invoke = AsyncMock(return_value=AsyncMock(value="result"))
+
+    mock_state = KernelProcessStepState(name="my_step", id="123", state=None)
+    mock_info = KernelProcessStepInfo(inner_step_type=AsyncMock(spec=type), state=mock_state, output_edges={})
+
+    step = LocalStep(
+        kernel=mock_kernel,
+        step_info=mock_info,  # type: ignore
+        event_namespace="ns",
+        step_state=mock_state,
+        factories={},
+        functions={},
+        inputs={"func": {"param": "ready_value"}},
+        initial_inputs={"func": {"param": None}},
+        initialize_task=True,
+    )
+
+    msg = LocalMessage(
+        source_id="source",
+        destination_id="dest",
+        function_name="func",
+        values={"param": "value"},
+    )
+
+    with pytest.raises(ProcessFunctionNotFoundException) as exc:
+        await step.handle_message(msg)
+    assert "Function func not found in plugin my_step" in str(exc.value)
+
+
+async def test_handle_message_emits_error_event_on_exception():
+    """Test handle_message emits an OnError event when the function invocation raises an exception."""
+    mock_kernel = AsyncMock(spec=Kernel)
+    mock_kernel.invoke = AsyncMock(side_effect=KernelException("error"))
+
+    mock_state = KernelProcessStepState(name="my_step", id="123", state=None)
+    mock_info = KernelProcessStepInfo(inner_step_type=AsyncMock(spec=type), state=mock_state, output_edges={})
+    mock_function = AsyncMock(spec=KernelFunction)
+    mock_function.name = "func"
+    mock_function.plugin_name = "test_plugin"
+
+    mock_function.metadata = AsyncMock(spec=KernelFunctionMetadata)
+
+    mock_function.metadata.name = "func"
+    mock_function.metadata.plugin_name = "test_plugin"
+    mock_function.metadata.is_prompt = False
+
+    step = LocalStep(
+        kernel=mock_kernel,  # type: ignore
+        step_info=mock_info,
+        event_namespace="ns",
+        step_state=mock_state,
+        factories={},
+        functions={"func": mock_function},
+        inputs={"func": {"param": "some_value"}},
+        initial_inputs={"func": {"param": None}},
+        initialize_task=True,
+    )
+
+    with patch.object(LocalStep, "emit_event") as mock_emit_event:
+        msg = LocalMessage(
+            source_id="source",
+            destination_id="dest",
+            function_name="func",
+            values={},
+        )
+
+        await step.handle_message(msg)
+
+        # The event name for error is "func.OnError"
+        assert mock_emit_event.await_args is not None
+        call_args = mock_emit_event.await_args.args
```

Comment:
Instead of peeking into mock args, should we assert that `emit_event` was awaited once and that the sent event’s `id` ends with `.OnError`?